students/AndrewMiotke/session04/trigrams.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+import random
+
+"""
+Solution to trigram problem
+"""
+
+def read_file(filename):
+    """
+    returns a list of words in the file
+    """
+    words = [] # creates empty list
+    with open(filename) as text:
+        for line in text:
+            words.extend(line.split())
+
+    return words
+
+
+def build_trigrams(word_list):
+    """
+    takes a list of words and makes a trigrams dict
+    """
+    trigrams = {}
+
+    for ind in range(len(word_list) - 2):
+        w1 = word_list[ind]
+        w2 = word_list[ind + 1]
+        w3 = word_list[ind + 2]
+        pair = (w1, w2)
+
+        if pair not in trigrams:
+            trigrams[pair] = [w3]
+        else:
+            trigrams[pair].append(w3)
+
+    return trigrams
+
+
+def create_sentence(trigram):
+    sentence = []
+    starting_key = random.choice(list(trigram.keys()))
+    sentence.extend([word for word in starting_key])
+    sentence.append(trigram[starting_key][random.choice(range(0, len(trigram[starting_key])))])
+
+    while len(sentence) < 15:
+        try:
+            key1, key2 = sentence[-2:]
+            sentence.append(trigram[(key1, key2)][random.choice(range(0, len(trigram[(key1, key2)])))])
+        except KeyError:
+            break
+
+    sentence[0] = sentence[0].capitalize()
+    if sentence[-1][-1] != ".":
```

Comment:
if you have periods in your work list -- you may wan tot let them break the sentences for you, randomly.